src/adapters/aol.js

```diff
@@ -1,183 +1,140 @@
-
-  // constants
-  var ADTECH_URI = 'https://secure-ads.pictela.net/rm/marketplace/pubtaglib/0_4_0/pubtaglib_0_4_0.js';
-  var ADTECH_BIDDER_NAME = 'aol';
-  var ADTECH_PUBAPI_CONFIG = {
-    pixelsDivId: 'pixelsDiv',
-    defaultKey: 'aolBid',
-    roundingConfig: [
-      {
-        from: 0,
-        to: 999,
-        roundFunction: 'tenCentsRound'
-      }, {
-        from: 1000,
-        to: -1,
-        roundValue: 1000
-      }
-    ],
-    pubApiOK: _addBid,
-    pubApiER: _addErrorBid
+const utils = require('../utils.js');
+const ajax = require('../ajax.js').ajax;
+const bidfactory = require('../bidfactory.js');
+const bidmanager = require('../bidmanager.js');
+
+const AolAdapter = function AolAdapter() {
+
+  const pubapiTemplate = template`${'protocol'}://${'host'}/pubapi/3.0/${'network'}/${'placement'}/${'pageid'}/${'sizeid'}/ADTECH;v=2;cmd=bid;cors=yes;alias=${'alias'}${'bidfloor'};misc=${'misc'}`;
+  const BIDDER_CODE = 'aol';
+  const SERVER_MAP = {
+    us: 'adserver-us.adtech.advertising.com',
+    eu: 'adserver-eu.adtech.advertising.com',
+    as: 'adserver-as.adtech.advertising.com'
   };
 
-  var bids;
-  var bidsMap = {};
-  var d = window.document;
-  var h = d.getElementsByTagName('HEAD')[0];
-  var dummyUnitIdCount = 0;
-
-  /**
-   * @private create a div that we'll use as the
-   * location for the AOL unit; AOL will document.write
-   * if the div is not present in the document.
-   * @param {String} id to identify the div
-   * @return {String} the id used with the div
-   */
-  function _dummyUnit(id) {
-    var div = d.createElement('DIV');
-
-    if (!id || !id.length) {
-      id = 'ad-placeholder-' + (++dummyUnitIdCount);
-    }
-
-    div.id = id + '-head-unit';
-    h.appendChild(div);
-    return div.id;
+  function template(strings, ...keys) {
+    return function(...values) {
+      let dict = values[values.length - 1] || {};
+      let result = [strings[0]];
+      keys.forEach(function(key, i) {
+        let value = Number.isInteger(key) ? values[key] : dict[key];
+        result.push(value, strings[i + 1]);
+      });
+      return result.join('');
+    };
   }
 
-  /**
-   * @private Add a succesful bid response for aol
-   * @param {ADTECHResponse} response the response for the bid
-   * @param {ADTECHContext} context the context passed from aol
-   */
-  function _addBid(response, context) {
-    var bid = bidsMap[context.alias];
-    var cpm;
-
-    if (!bid) {
-      utils.logError('mismatched bid: ' + context.placement, ADTECH_BIDDER_NAME, context);
-      return;
-    }
+  function _buildPubapiUrl(bid) {
+    const params = bid.params;
+    const serverParam = params.server;
+    let regionParam = params.region || 'us';
+    let server;
 
-    cpm = response.getCPM();
-    if (cpm === null || isNaN(cpm)) {
-      return _addErrorBid(response, context);
+    if (!SERVER_MAP.hasOwnProperty(regionParam)) {
+      console.warn(`Unknown region '${regionParam}' for AOL bidder.`);
```

Comment:
`console.warn` doesn't play nice in IE 9. Please use `utils.logWarn` instead for safety.